src/resdk/collection_tables.py

```diff
@@ -0,0 +1,290 @@
+            with BytesIO() as f:
+                f.write(response.content)
+                f.seek(0)
+                df_ = pd.read_csv(f, sep='\t', compression='gzip')
+                df_ = df_.set_index('Gene').T
+                df_.index = [ts.sample_slug]
+                df_list.append(df_)
+
+        df = pd.concat(df_list, axis=0).sort_index()
+        df.index.name = 'sample_slug'
+        df.columns.name = ''
+        df = self._gene_mapping(df)
+        return df
+
+    def _gene_mapping(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Transform columns of the given table from ensemble ids to gene symbols."""
+        genes = df.columns.tolist()
+        gene_map = self._gene_map(genes)
+        df.columns = df.columns.map(gene_map)
+        return df
+
+    def _get_metadata(self) -> pd.DataFrame:
+        """Flatten samples JSON metadata into a table."""
+        meta = pd.json_normalize([ts.metadata for ts in self._table_samples])
+        meta = meta.set_index('sample_slug').sort_index()
+        return meta
+
+    def _get_table_samples(self):
+        """Fetch expressions files urls and sample metadata."""
+        table_samples = []
+        for sample in self.collection.samples.iterate():
+            # get sample metadata from description
+            metadata = sample.descriptor
+            metadata['sample_slug'] = sample.slug
+
+            # get expressions data objects
+            exp_data = sample.get_expression()
+
+            # get expression files urls
+            expression_files = {}
+            for exp_type in EXPRESSION_TYPES:
+                exp_files = exp_data.files(field_name=exp_type)
+                assert len(exp_files) == 1, "Multiple expression files with same expression type!"
+                exp_file = exp_files[0]
+                exp_file_url = "{}/{}".format(exp_data.id, exp_file)
+                expression_files[exp_type] = exp_file_url
+
+            table_samples.append(TableSample(
+                sample_slug=sample.slug,
+                expression_files=expression_files,
+                metadata=metadata
+            ))
+        return table_samples
+
+    def _get_gene_map(self, ensembl_ids: List[str]) -> dict:
+        """Fetch gene mapping from resolwe server."""
+        sublists = [
+            ensembl_ids[i: i + CHUNK_SIZE]
+            for i in range(0, len(ensembl_ids), CHUNK_SIZE)
+        ]
+        species = self._table_samples[0].metadata['general']['species']
+        gene_map = {}
+        for sublist in sublists:
+            features = self.resolwe_obj.feature.filter(species=species,
+                                                       feature_id__in=sublist)
+            gene_map.update({f.feature_id: f.name for f in features})
+        return gene_map
+
+    def _get_modified(self) -> datetime:
+        """Get the newest modification date out of all sample in the collection."""
+        newest_modified = reduce(max, map(lambda s: s.modified,
```

Comment:
Any reason why not just:
`max([s.modified for s in self.collection.samples.iterate()])`